src/web/nextui/next.config.js

```diff
@@ -1,13 +1,29 @@
-const outputType = process.env.NEXT_PUBLIC_PROMPTFOO_BUILD_STANDALONE_SERVER
-  ? 'standalone'
-  : 'export';
+const API_PORT = process.env.API_PORT || '15500';
+
+if (process.env.NODE_ENV === 'development') {
+  process.env.NEXT_PUBLIC_PROMPTFOO_REMOTE_API_BASE_URL =
+    process.env.PROMPTFOO_REMOTE_API_BASE_URL || `http://localhost:${API_PORT}`;
+  process.env.NEXT_PUBLIC_PROMPTFOO_SHARE_API_URL = `http://localhost:${API_PORT}`;
+} else {
+  if (process.env.NEXT_PUBLIC_HOSTED) {
+    process.env.NEXT_PUBLIC_PROMPTFOO_SHARE_API_URL =
+      process.env.PROMPTFOO_REMOTE_API_BASE_URL || '';
+  } else {
+    process.env.NEXT_PUBLIC_PROMPTFOO_APP_SHARE_URL = 'https://app.promptfoo.dev';
+    process.env.NEXT_PUBLIC_PROMPTFOO_SHARE_API_URL = 'https://api.promptfoo.dev';
+  }
```

Comment:
If it's a hosted server (which we set in the docker build) the share API will be an empty string. Since the web app and API are hosted from the same url. If it's not hosted it means it was built for the CLI/NPM package so we want the share url's to be our public ones.